### PR TITLE
ingress how to

### PR DESCRIPTION
**Pull request by sjletcher**

### CODE REVIEW

**Comment by jits** [comment]
What's the scope of this how to do? When the user sees this, are they expected to already be familiar with _what_ ingress is? Or should we provide more context on what ingress is and why they might want to use it?

File: how-to-docs/ingress.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+
+  
+ ```---
+ apiVersion: extensions/v1beta1
+ kind: Ingress
+ metadata:
+   annotations:
+     # kubernetes.io/tls-acme: "true"
```
Comment (jits) [inline how-to-docs/ingress.md]: Can remove this.

File: how-to-docs/ingress.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# Using Ingress
+
+An Ingress is a type of Kubernetes resource that allows you to expose your services outside the cluster it gets deployed and managed exactly like other Kube resources.
+
+Our ingress setup offers two different ingresses based on how restrictively you want to expose your services:
+- internal - only people within the VPN can access services
+- external - anyone with internet access can access services
+
+The annotation "kubernetes.io/ingress.class: "nginx-internal"" is used to specify whether the ingress is internal or external.
```
Comment (jits) [inline how-to-docs/ingress.md]: Can say:

```md
The annotation `kubernetes.io/ingress.class` is used to specify whether the ingress is internal or external.
```

**Comment by vinn946** [comment]
You might also want to add a link to your specific how to on how-to-docs/README

File: how-to-docs/ingress.md [hunk 3]
```diff
@@ -0,0 +1,36 @@
+# Using Ingress
+
+An Ingress is a type of Kubernetes resource that allows you to expose your services outside the cluster it gets deployed and managed exactly like other Kube resources.
```
Comment (jits) [inline how-to-docs/ingress.md]: Needs a full stop after `outside the cluster`.

File: how-to-docs/ingress.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+Our ingress setup offers two different ingresses based on how restrictively you want to expose your services:
+- internal - only people within the VPN can access services
+- external - anyone with internet access can access services
+
+The annotation ```kubernetes.io/ingress.class: "nginx-internal"``` is used to specify whether the ingress is internal. (```kubernetes.io/ingress.class: "nginx-external"``` is used for an external ingress.)
```
Comment (jits) [inline how-to-docs/ingress.md]: Note you can also just use a single backtick when you have code inline like this. (I didn't even know 3 backticks were allowed inline! :))

File: how-to-docs/ingress.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+       http:
+         paths:
+           - backend:
+               serviceName: myapp
+               servicePort: 8000
+             path: /
+   tls:
+     - hosts:
+         - "myapp.myproject.homeoffice.gov.uk"
+       secretName: myapp-github-internal-tls
```
Comment (jits) [inline how-to-docs/ingress.md]: Once we've figured out how we will offer TLS certs generation to users, we'll need to update this doc. I know @gambol99 did put something in place (which can be triggered via another annotation), but I'm not sure if that's "production ready".

**Pull request merged**